draft-ietf-oauth-transaction-tokens.md

```diff
@@ -626,6 +635,9 @@ The authorization model within a trust domain boundary is most often quite diffe
 ## Identifying Call Chains
 A Txn-token typically represents the call-chain of workloads necessary to complete a logical function initiated by an external or internal workload. The `txn` claim in the Txn-token provides a unique identifier that when logged by the TTS and each subsequent workload can provide both discovery and auditability of successful and failed transactions. It is therefore strongly RECOMMENDED to use an identifier, unique within the trust domain, for the `txn` value.
 
+## Workload Configuration Protection
+A workload may be configured to access more than one Transaction Token Service to ensure redundency or reduce latency for transaction token requests. The workload configuration should be protected against unauthorized addition or removal of the Transaction Token Services that it may access. An attacker may perform a denial of service attack or degrade the performance of a system by removing Transaction Token Services from the workload configuration. Alternatively, an attacker may add its own Transaction Token Service to the configuration, resulting in the workload sending sensitive information such as Access Tokens to a Transaction token Service under the attacker's control.
```

Comment:
I can see cases where a transaction token service only issues tokens for a single domain, but I can also imagine that one is shared by two or more domains, especially if it is a more centralised model and it is attached to a Authorisation Server that services an entire enterprise, and must issue txn-tokens to many trust domains. 

I do like the simplifications of a single transaction token service per trust domain though, but I wonder if that is how it will get deployed in practice.

What else breaks in the spec if we don't make this assumption?

Reply:
Updated to reflect a single logical transaction token service that may have multiple instances.